pkg/main.go

```diff
@@ -222,13 +192,68 @@ func main() {
 			output4 := strings.Join(non_rpm_not_fixed_status, "")
 			Information = fmt.Sprintf("%v%v%v%v", output1, output2, output3, output4)
 		}
+		// Summary := new(Output)
+		var summary Output
+		summary.CVE = cve_id
+		summary.Component = line_data[1]
+		summary.Severity = result.Severity
+		summary.Information = Information
+
+		results <- summary
+	}
+}
+
+func main() {
+	// Verification if provided file with data exists or not
+	if len(os.Args) < 2 {
+		fmt.Println("File to check is not provided")
+		os.Exit(1)
+	}
+	file_name := os.Args[1]
+	if fileExist(file_name) {
+		fmt.Printf("Checking the %v\n", file_name)
+	} else {
+		fmt.Println("file to check is incorrect or not exist")
+		os.Exit(1)
+	}
+
+	// Opening the file
+	file, err := os.Open(file_name)
+	if err != nil {
+		fmt.Println("failed opening file")
+	}
+	defer file.Close()
+
+	data := csv.NewReader(file)
+
+	// Checking the data
+	jobs := make(chan []string)
+	results := make(chan Output, 50) // guess
```

Comment:
Why 50?

Reply:
Can be anything you like, ideally it would be the length of the results you're returning. But we don't know that yet here, so allocated 50 result objects so it's one less thing for the loop to do. It'll expand it if required, just means the loop isn't growing it, till 50 objects exist. I figured 50 is a good start.

Reply:
Ah, that makes sense!